src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java

```diff
@@ -251,6 +258,156 @@ public void should_return_a_200_with_empty_search_results_user_jurisdiction_perm
     }
 
 
+    @Test
+    public void should_return_a_200_with_search_results_and_sorted_by_case_id_descendant() {
+        String[] taskStates = {TaskState.UNASSIGNED.value(), TaskState.ASSIGNED.value(), TaskState.CONFIGURED.value()};
+
+        List<Map<String, String>> tasksCreated = createMultipleTasksWithDifferentTaskStates(taskStates);
+
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(
+            asList(
+                new SearchParameter(JURISDICTION, SearchOperator.IN, singletonList("IA")),
+                new SearchParameter(LOCATION, SearchOperator.IN, singletonList("765324")),
+                new SearchParameter(STATE, SearchOperator.IN, asList("unassigned", "assigned"))
+            ),
+            singletonList(new SortingParameter(CASE_ID, DESCENDANT))
+        );
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchTaskRequest,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.id", hasItems(tasksCreated.get(0).get("taskId"), tasksCreated.get(1).get("taskId")))
+            .body("tasks.case_id", hasItems(tasksCreated.get(0).get("caseId"), tasksCreated.get(1).get("caseId")))
+            .body("tasks.task_state", everyItem(either(is("unassigned")).or(is("assigned"))))
+            .body("tasks.location", everyItem(equalTo("765324")))
+            .body("tasks.jurisdiction", everyItem(equalTo("IA")));
+
+        List<String> jsonResponse = result.jsonPath().getList("tasks.case_id");
+        assertTrue(Ordering.natural().reverse().isOrdered(jsonResponse));
```

Comment:
This ordering approach looks good to me. We might want to write a simple unit test to cover this line is isolation and confirm it works as we expect.

Reply:
good idea I'll add an extra unit test for confidence